lib/spack/spack/filesystem_view.py

```diff
@@ -67,7 +67,11 @@ def view_copy(src, dst, view, spec=None):
 
     Use spec and view to generate relocations
     """
-    shutil.copy2(src, dst)
+    if os.path.isdir(src):
+        shutil.copytree(src, dst, symlink=True)
```

Comment:
More generally, I think @haampie is suggesting changing this to something like:

```suggestion
    if os.path.islink(src):
        os.symlink(src, dst)
```

If that works, it may take care of the issue without needing to traverse the directory.

Reply:
It's not enough for absolute symlinks into the prefix dir, they have to be relocated

Reply:
If they're into the prefix dir, they will have been relocated by being crawled through the non-symlink directory anyway right? Ah, but the target will be wrong because the src is still in the prefix, is that what you mean?